AwsCryptographicMaterialProviders/dafny/AwsCryptographyKeyStore/Model/Storage.smithy

```diff
@@ -0,0 +1,203 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+namespace aws.cryptography.keyStore
+
+
+//= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#type
+//= type=implication
+//# A union that MUST hold the following three options
+//#
+//#- ActiveHierarchicalSymmetricVersion [HierarchicalSymmetricVersion](#hierarchicalsymmetricversion)
+//#- HierarchicalSymmetricVersion [HierarchicalSymmetricVersion](#hierarchicalsymmetricversion)
+//#- ActiveHierarchicalSymmetricBeacon
+union BranchKeyType {
+  ActiveHierarchicalSymmetricVersion: String,
+  //= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#hierarchicalsymmetricversion
+  //= type=implication
+  //# A structure that MUST have one member,
+  //# the UTF8 Encoded value of the version of the branch key.
+  HierarchicalSymmetricVersion: String,
+  ActiveHierarchicalSymmetricBeacon: ActiveHierarchicalSymmetricBeacon
+}
+
+structure ActiveHierarchicalSymmetricBeacon {}
+
+
+//= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#encryptedhierarchicalkey
+//= type=implication
+//# This structure MUST include all of the following fields:
+//#
+//#- [BranchKeyId](./structures.md#branch-key-id)
+//#- [Type](#type)
+//#- CreateTime: Timestamp in ISO 8601 format in UTC, to microsecond precision.
+//#- KmsArn: The AWS KMS Key ARN used to protect the CiphertextBlob value.
+//#- [EncryptionContext](./structures.md#encryption-context-3)
+//#- CiphertextBlob: The encrypted binary for the hierarchical key.
+structure EncryptedHierarchicalKey {
+  @required
+  Identifier: String,
+
+  @required
+  Type: BranchKeyType,
+
+  @required
+  CreateTime: String,
+
+  @required
+  KmsArn: String,
+
+  @required
+  EncryptionContext: EncryptionContextString,
+
+  @required
+  CiphertextBlob: Blob,
+}
+
+map EncryptionContextString {
+  key: String,
+  value: String,
+}
+
+@aws.polymorph#extendable
+resource EncryptedKeyStore {
+
+//= aws-encryption-sdk-specification/framework/key-store/encrypted-key-store.md#interface
+//= type=implication
+//# The EncryptedKeyStore MUST support the following operations:
+//#
+//#- [WriteNewKeyToStore](#writenewkeytostore)
+//#- [WriteNewBranchKeyVersionToKeystore](#writenewbranchkeyversiontokeystore)
+//#- [GetEncryptedActiveBranchKey](#getencryptedactivebranchkey)
+//#- [GetEncryptedBranchKeyVersion](#getencryptedbranchkeyversion)
+//#- [GetEncryptedBeaconKey](#getencryptedbeaconkey)
+//#- [GetTableName](#gettablename)
+  operations: [
+    WriteNewKeyToStore,
+    WriteNewBranchKeyVersionToKeystore,
+    GetEncryptedActiveBranchKey,
+    GetEncryptedBranchKeyVersion,
+    GetEncryptedBeaconKey,
+    GetTableName,
```

Comment:
I wish we could make this a more generic method. 
I.e DescribeStorage. 

Imagine an S3 bucket or some other storage that does not use tables.

Reply:
@lucasmcdonald3  and I talked about this.
This value gets returned to the actual caller as "table name" because no one ever calls this directly, only the Key store. Further it will always be a "table like structure".

Reply:
Sure... let me try a different sales approach.
DynamoDB, the service, has `Get*` operations.
They also have `Describe*` operations.

Both operations return data.
But the `Get*` operations return customer data,
while the `Describe*` operations return (potentially customer determined) metadata. 

I suggest that `GetTableName` is returning metadata/configuration data. 
Thus, 
I suggest that is better labeled `Describe`.

Reply:
Changed to `GetKeyStorageInfo`